pim-tool-back/src/main/java/vn/elca/training/web/ProjectController.java

```diff
@@ -1,32 +1,67 @@
 package vn.elca.training.web;
 
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+
 import vn.elca.training.model.dto.ProjectDto;
+import vn.elca.training.model.entity.Project;
 import vn.elca.training.service.ProjectService;
-import vn.elca.training.util.ApplicationMapper;
-
-import java.util.List;
-import java.util.stream.Collectors;
 
 /**
- * @author gtn
- *
+ * @author thomas.dang
+ * 
  */
 @RestController
-@RequestMapping("/projects")
+@RequestMapping("/project")
 public class ProjectController extends AbstractApplicationController {
 
     @Autowired
+    @Qualifier("firstDummyProjectServiceImpl")
     private ProjectService projectService;
 
-    @GetMapping("/search")
-    public List<ProjectDto> search() {
-        return projectService.findAll()
-                .stream()
-                .map(mapper::projectToProjectDto)
-                .collect(Collectors.toList());
+    @GetMapping("/id/{id}")
+    public ResponseEntity<ProjectDto> searchById(@PathVariable long id) {
+        Logger.getLogger(ProjectController.class.getName())
```

Comment:
[N] Logger can be declared as a singleton at class level. In method, we can just use that singleton for simplicity.